integration/querydsl/expressions/src/main/java/com/blazebit/persistence/querydsl/BlazeCriteriaBuilderRenderer.java

```diff
@@ -475,10 +475,13 @@ private void renderDistinct(QueryMetadata metadata, DistinctBuilder<?> criteriaB
                 }
             } else if (target instanceof Path<?>) {
                 Path<?> entityPath = (Path<?>) target;
-                if (alias == null) {
-                    alias = entityPath.getMetadata().getName();
-                }
                 boolean entityJoin = entityPath.getMetadata().isRoot();
+                String renderedExpression = renderExpression(entityPath);
+
+                // While this looks suspicious, this is actually in line with Querydsl's default behaviour in JPQLSerializer.handleJoinTarget
+                if (alias == null && entityJoin) {
+                    alias = renderedExpression;
```

Comment:
Not sure what `entityJoin` means exactly in this context, but using an entity name as alias is asking for trouble as you can see here: https://github.com/Blazebit/blaze-persistence/issues/1450

Reply:
In Querydsl it is conventional, because an EntityPath is both a Path and a From:

```java
.from(QEntity.entity)
```

Renders:

```sql
FROM Entity entity
```

And:

```java
.from(QEntity.entity)
.innerJoin(QOtherEntity.otherEntity).on(QOtherEntity.otherEntity.id.eq(1))
```

Renders:

```sql
FROM Entity entity INNER JOIN OtherEntity otherEntity on otherEntity.id = 1
```

It's also conventional to create Q-class instances with a unique name:

```java
QEntity alias = new QEntity("alias");
.from(alias)
```

Renders:

```sql
FROM Entity alias
```

So in order to support these syntaxes, we need to consider the alias in the case it is a root path (in which case it is also an entity join).

However, the naming strategy for the "default" alias (`QEntity.entity`) is configurable. It is also possible to disable the default alias altogether. So I think there are plenty of ways to work around eventual issues, and these are issues that Querydsl users had to guard themselves against anyway.

Reply:
Sounds good, thanks for the explanation!